### PR TITLE
Record correct specialized stride info when fake tensors disabled

### PR DESCRIPTION
**Pull request by ezyang**
Fixes https://github.com/pytorch/torchdynamo/issues/945

### CODE REVIEW

File: torchdynamo/variables/tensor.py [hunk 1]
```diff
@@ -154,7 +154,8 @@ def context():
 
             # tensor subclasses will not be converted to FakeTensors and need to be cloned
             if not use_fake_tensors or not isinstance(example_value, FakeTensor):
-                example_value = clone_tensor(example_value)
+                # NB: ensure strides are preserved
+                example_value = clone_input(example_value)
```
Comment (anijain2305) [inline torchdynamo/variables/tensor.py:158]: `clone_input` does not copy the gradient of the tensor.
I was thinking of changing `clone_tensor` in utils.py to use `clone_input` instead of using `torch.clone()`.

Comment (ezyang) [reply torchdynamo/variables/tensor.py:158]: Cloning the gradient here seems unnecessary. Here are the uses of the subsequent `example_value`:

```
            proxy.node.meta["example_value"] = example_value
            specialized_props = cls.specialize(example_value)
            if use_fake_tensors and isinstance(example_value, FakeTensor):
                specialized_props["class_type"] = (
                    torch.nn.Parameter if is_parameter else torch.Tensor
                )
```

Specialize does not check the grad field of the tensor. So the only way the grad could matter is if the proxy.node.meta expected the example value to have the grad field accurately set. But I could easily define the contract for this meta that it does not have to accurately represent grad...

**Comment by ezyang** [comment]
It looks like `clone_input` does not work in enough situations. Need a stronger cloner...

**Comment by ezyang** [comment]
I fixed the quantization problem, please review

**Review by jansel — APPROVED** [review]
